lib/modem_info/modem_info_json.c

```diff
@@ -97,67 +57,190 @@ static int json_add_str(cJSON *parent, const char *str, const char *item)
 	return json_add_obj(parent, str, json_str);
 }
 
-int modem_info_json_string_get(char *buf)
+static int json_add_data(struct lte_param *param, cJSON *json_obj)
 {
-	u16_t param_value;
-	size_t len = 0;
-	size_t total_len = 0;
-	int ret = 0;
-	char data_buffer[MODEM_INFO_MAX_RESPONSE_SIZE] = {0};
 	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
-	cJSON *data_obj = cJSON_CreateObject();
 	enum at_param_type data_type;
+	size_t total_len = 0;
+	size_t ret;
```

Comment:
`size_t` is an unsigned int, but is later checked if it is negative, so this should be changed to `int`.

Reply:
Also the type for `data_type` variable should reconsidered